frontend/packages/dev-console/src/components/import/serverless/serverless-utils.ts

```diff
@@ -59,10 +59,10 @@ export const getAllOtherDomainMappingInUse = (
     .filter((d) => hasOtherKsvcDomainMappings([d]))
     .map((dm) => {
       const selectedDomain = data?.find(
-        (d) => d.metadata.name === removeKsvcInfoFromDomainMapping(dm),
+        (d) => d?.metadata?.name === removeKsvcInfoFromDomainMapping(dm),
       );
       return selectedDomain && selectedDomain.spec?.ref?.name !== serviceName
         ? selectedDomain
         : null;
     })
-    .filter((d) => !!d);
+    .filter((d): d is K8sResourceKind => !!d);
```

Comment:
i am not sure if this would correctly determine whether some data is a k8sresourcekind